kasa/module.py

```diff
@@ -37,6 +37,12 @@ def query(self):
     def data(self):
         """Return the module specific raw data from the last update."""
 
+    def _initialize_features(self):
+        """Initialize features after the initial update.
+
+        This can be implemented if features depend on module query responses.
+        """
+
     def _add_feature(self, feature: Feature):
         """Add module feature."""
```

Comment:
I think this would be more intuitive if all the modules moved their feature initialization out of the constructors and into this method. Then there would only be one path the initializing them which can always take place after the initial update.

Reply:
Yeah, that makes sense. Although it's better to do that refactoring in a separate PR that only involves that change, and do it after some of those other open PRs gets merged, as this will affect all of them.